crates/trigger/src/lib.rs

```diff
@@ -1,118 +1,124 @@
-use std::{error::Error, path::PathBuf};
+use std::{error::Error, marker::PhantomData, path::PathBuf};
 
-use anyhow::{Context, Result};
+use anyhow::Result;
 use async_trait::async_trait;
 use spin_engine::{
     io::FollowComponents, Builder, Engine, ExecutionContext, ExecutionContextConfiguration,
 };
-use spin_manifest::{Application, ApplicationTrigger, ComponentMap, TriggerConfig};
+use spin_manifest::{Application, ApplicationTrigger, TriggerConfig};
 
-/// The trigger
+pub mod cli;
 #[async_trait]
-pub trait Trigger: Sized {
-    /// data
-    type ContextData: Default + 'static;
-    /// trigger configuration
-    type Config;
-    /// component configuration
-    type ComponentConfig;
-    /// runtime configuration
-    type ExecutionConfig;
+pub trait TriggerExecutor: Sized {
+    type GlobalConfig;
+    type TriggerConfig;
+    type RunConfig;
+    type RuntimeContext: Default + 'static;
```

Comment:
This renaming is aiming towards a future manifest rewrite a la https://hackmd.io/@lann/SyfiEmEHc